### PR TITLE
[Merged by Bors] - Add `lighthouse db` command

### PR DESCRIPTION
**Pull request by michaelsproul**
## Proposed Changes

Add a `lighthouse db` command with three initial subcommands:

- `lighthouse db version`: print the database schema version.
- `lighthouse db migrate --to N`: manually upgrade (or downgrade!) the database to a different version.
- `lighthouse db inspect --column C`: log the key and size in bytes of every value in a given `DBColumn`.

This PR lays the groundwork for other changes, namely:

- Mark's fast-deposit sync (https://github.com/sigp/lighthouse/pull/2915), for which I think we should implement a database downgrade (from v9 to v8).
- My `tree-states` work, which already implements a downgrade (v10 to v8).
- Standalone purge commands like `lighthouse db purge-dht` per https://github.com/sigp/lighthouse/issues/2824.

## Additional Info

I updated the `strum` crate to 0.24.0, which necessitated some changes in the network code to remove calls to deprecated methods.

Thanks to @winksaville for the motivation, and implementation work that I used as a source of inspiration (https://github.com/sigp/lighthouse/pull/2685).

### CODE REVIEW

**Comment by winksaville** [comment]
I'm happy to see this, great work!

File: lighthouse/src/main.rs [hunk 1]
```diff
@@ -485,7 +486,16 @@ fn run<E: EthSpec>(
 
         // Exit as soon as account manager returns control.
         return Ok(());
-    };
+    }
+
+    if let Some(sub_matches) = matches.subcommand_matches("database_manager") {
```
Comment (realbigsean) [inline lighthouse/src/main.rs]: Could use `database_manager::CMD` here

File: database_manager/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+                .default_value("sizes")
+                .possible_values(InspectTarget::VARIANTS),
+        )
+}
+
+pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
+    App::new(CMD)
+        .visible_aliases(&["db"])
+        .setting(clap::AppSettings::ColoredHelp)
+        .about("")
```
Comment (realbigsean) [inline database_manager/src/lib.rs]: Is this meant to be empty?

File: database_manager/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+pub fn display_db_version<E: EthSpec>(
+    client_config: ClientConfig,
+    runtime_context: &RuntimeContext<E>,
+    log: Logger,
+) -> Result<(), Error> {
+    let spec = runtime_context.eth2_config.spec.clone();
+    let hot_path = client_config.get_db_path();
+    let cold_path = client_config.get_freezer_db_path();
+
+    let mut version = CURRENT_SCHEMA_VERSION;
```
Comment (realbigsean) [inline database_manager/src/lib.rs:117]: Might be useful to log out `CURRENT_SCHEMA_VERSION` too?

**Review by realbigsean — COMMENTED** [review]
Couple tiny things, but this is pretty sweet!

File: database_manager/src/lib.rs (hunk 2 above)
Comment (michaelsproul) [reply database_manager/src/lib.rs]: No! Good catch!

File: database_manager/src/lib.rs (hunk 3 above)
Comment (michaelsproul) [reply database_manager/src/lib.rs:117]: I've added a conditional log in 6dc57a0cbe956d3aa9fb782cae3ec5c61e515474

**Review by realbigsean — APPROVED** [review]
LGTM

**Comment by michaelsproul** [comment]
Thanks!

bors r+

**Comment by michaelsproul** [comment]
bors r-

gonna batch

**Pull request closed without merging**